torch/csrc/api/include/torch/nn/functional/activation.h

```diff
@@ -65,6 +66,95 @@ inline Tensor rrelu(Tensor& input, const RReLUOptions& options, bool training=fa
   }
 }
 
+inline Tensor multi_head_attention_forward(
+  const Tensor& query,
+  const Tensor& key,
+  const Tensor& value,
+  int64_t embed_dim_to_check,
+  int64_t num_heads,
+  const Tensor& in_proj_weight,
+  const Tensor& in_proj_bias,
+  const c10::optional<Tensor>& bias_k,
+  const c10::optional<Tensor>& bias_v,
+  bool add_zero_attn,
+  double dropout_p,
+  const Tensor& out_proj_weight,
+  const Tensor& out_proj_bias,
+  bool training = true,
+  const c10::optional<Tensor>& key_padding_mask = c10::nullopt,
+  bool need_weights = true,
+  const c10::optional<Tensor>& attn_mask = c10::nullopt,
+  bool use_separate_proj_weight = false,
+  const c10::optional<Tensor>& q_proj_weight = c10::nullopt,
+  const c10::optional<Tensor>& k_proj_weight = c10::nullopt,
+  const c10::optional<Tensor>& v_proj_weight = c10::nullopt,
+  const c10::optional<Tensor>& static_k = c10::nullopt,
+  const c10::optional<Tensor>& static_v = c10::nullopt) {
```

Comment:
I feel that we should have another options called `MultiheadAttentionForwardOptions` to represent all the optional arguments for this functional, and also use that options in the module's forward method to represent the optional arguments. Do you think it would be a good idea?